src/Repository/ActivityRepository.php

```diff
@@ -52,26 +55,72 @@ public function save(Activity $activity): void
         $em->flush();
     }
 
-    public function getAvailableActivities(User $user): QueryBuilder
-    {
+    public function getAvailableActivities(
+        ActivityListSort $activityListSort,
+        ActivityListFilter $activityListFilter,
+        User $user
+    ): QueryBuilder {
         $queryBuilder = $this->createQueryBuilder('activity');
         $queryBuilder
-            ->select('activity')
+            ->select('DISTINCT activity')
             ->leftJoin('activity.activityUsers', 'activityUsers')
             ->where('activity.public = 1')
```

Comment:
For these 3 `where/orWhere` you should be using https://www.doctrine-project.org/api/orm/latest/Doctrine/ORM/Query/Expr/Orx.html in order to exclude collision with the next conditions.